lib/asker.js

```diff
@@ -246,6 +246,41 @@ Request.prototype._processOptions = function(options) {
     return opts;
 };
 
+Request.prototype._createAgent = function(agentOpts) {
+    var options = this.options,
+        defaultOpts = {
+            name: undefined,
+            protocol: options.protocol
+        };
+
+    // @note we have to pass TLS options to Agent explicitly because Node.js<0.12 doesn't do it
```

Comment:
This options can't be used if `defaultOpts`, as `Object.assign` copy all enumerable keys even if they are `undefined`. In this case it may lead to unwanted sideeffects.